e2e/allergy/allergy-detail/allergy-dialog.po.ts

```diff
@@ -0,0 +1,37 @@
+import { BasePage } from '../../common/components/base-page';
+import { by } from 'protractor';
+import { TabService } from '../../common/components/tab.service';
+
+export class AllergyDetailPage extends BasePage {
+
+    constructor() {
+        super('allergy-dialog');
+    }
+
+    public getEnableSwitch() {
+        return this.getObjectById('AllergyEnableSwitch')
+            .element(by.tagname('input'));
+    }
+
+    public getNameInput() {
+        return this.getObjectById('AllergyNameInput');
+    }
+
+    public getSignsInput() {
+        return this.getObjectById('AllergySignsInput');
+    }
+
+    public getSymptomsInput() {
+        return this.getObjectById('AllergySymptomsInput');
+    }
+
+    public getAllButtons() {
```

Comment:
getAllButtons() is a generic method for all the Page Objects and it may be at the Base Page file.